addons/account/models/account_journal_dashboard.py

```diff
@@ -543,24 +562,7 @@ def _fill_sale_purchase_dashboard_data(self, dashboard_data):
         query_results_to_pay = {}
         late_query_results = {}
         for journal_type, journals in [('sale', sale_journals), ('purchase', purchase_journals)]:
-            if not journals:
-                continue
-            query, params = journals._get_open_sale_purchase_query(journal_type).select(
-                SQL("account_move_line.journal_id"),
-                SQL("account_move_line.company_id"),
-                SQL("account_move_line.currency_id AS currency"),
-                SQL("account_move_line.date_maturity < %s AS late", fields.Date.context_today(self)),
-                SQL("SUM(account_move_line.amount_residual) AS amount_total_company"),
-                SQL("SUM(account_move_line.amount_residual_currency) AS amount_total"),
-                SQL("COUNT(*)"),
-            )
-            query += " GROUP BY company_id, journal_id, currency, late"
-            self.env.cr.execute(query, params)
-            query_result = group_by_journal(self.env.cr.dictfetchall())
-            for journal in journals:
-                query_results_to_pay[journal.id] = query_result[journal.id]
-                late_query_results[journal.id] = [r for r in query_result[journal.id] if r['late']]
```

Comment:
Sorry, I hadn't realized yesterday we could do this - but I think there's a simpler way than factorizing the `_retrieve_query_params` method. 
We could change the signature of the `_get_open_sale_purchase_query` method so that it returns both the `Query` object (containing the FROM and the WHERE), *and* a list of sql expressions to SELECT. 

And, you can make it return one sql expression called `to_pay` in the SELECT, which says whether the amount is to pay or not.

Then, when computing `query_results_to_pay[journal.id]`, you can do this:
```py
                query_results_to_pay[journal.id] = [r for r in query_result[journal.id] if r['to_pay']]
```